app/views/RoomView/index.js

```diff
@@ -159,9 +159,20 @@ class RoomView extends React.Component {
 
 	componentDidMount() {
 		this.mounted = true;
+		const { room } = this.state;
+		const { navigation } = this.props;
+		if (room && room.observe) {
+			this.roomObservable = room.observe();
+			this.subscription = this.roomObservable
+				.subscribe((changes) => {
+					if (this.mounted) {
+						navigation.setParams({ title: changes.name });
```

Comment:
Most of this work is done here: https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/views/RoomView/index.js#L343

The only thing is missing is a condition testing if the title changed and this `navigation.setParams()`.

Reply:
@diegolmello Sorry, because of earlier mistake, this Pr got messed up too, so I raised the new PR with all the necessary changes